### PR TITLE
[Substrait to Velox] Add support for Array type

### PR DESCRIPTION
**Pull request by leoluan2009**
Add support for Array type when Substrait to Velox

### CODE REVIEW

**Comment by leoluan2009** [comment]
@mbasmanova  can you help review this?

File: velox/substrait/TypeUtils.cpp [hunk 1]
```diff
@@ -90,17 +90,23 @@ TypePtr toVeloxType(const std::string& typeName) {
       return VARCHAR();
     case TypeKind::VARBINARY:
       return VARBINARY();
+    case TypeKind::ARRAY: {
+      auto fieldTypes = getTypesFromCompoundName(typeName);
+      VELOX_CHECK(
```
Comment (mbasmanova) [inline velox/substrait/TypeUtils.cpp:95]: Use VELOX_CHECK_EQ for better error messages.

File: velox/substrait/TypeUtils.cpp [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 namespace facebook::velox::substrait {
 namespace {
-std::vector<std::string_view> getRowTypesFromCompoundName(
+std::vector<std::string_view> getTypesFromCompoundName(
```
Comment (mbasmanova) [inline velox/substrait/TypeUtils.cpp:21]: The comment inside this function needs updating as well. It currently says "CompoundName is like ROW<BIGINT,DOUBLE>..."

**Review by mbasmanova — APPROVED** [review]
@leoluan2009 Nice addition. Some comments below.

**Review by mbasmanova — APPROVED** [review]
@leoluan2009 Looks great. Thank you for the contribution.

**Comment by leoluan2009** [comment]
@mbasmanova  thank you!

**Pull request closed without merging**